pkgs/top-level/all-packages.nix

```diff
@@ -12445,7 +12449,11 @@ with pkgs;
   # the withPackages wrapper available. In the final cross-compiled package set
   # however, targetPackages won't be populated, so we need to fall back to the
   # plain, cross-compiled compiler (which is only theoretical at the moment).
-  ghc = targetPackages.haskellPackages.ghc or haskell.compiler.ghc902;
+  ghc = dontRecurseIntoAttrs
```

Comment:
The old `targetPackages.haskellPackages.ghc or …` needs to be preserved, since it's not the same as `haskell.compiler.ghc902` (e.g. it has a `withPackages` method).

Reply:
I messed up by copying one too many line from the other change. Thanks.